refactor(semantic): refactor `check_binding_identifier`

overlookmotel opened the pull request:
Pure refactor. We had 2 `if` blocks following each other. 1st is `if strict_mode && ...` and the 2nd `if !strict_mode && ...`. Convert into an `if`/`else` instead.

camc314 reviewed (commented):
can we put this on top of the other stack? else I think there'll be merge conflicts.

https://github.com/oxc-project/oxc/pull/12298

overlookmotel commented:
No problem. I'll rebase this now the other stack has been merged. I didn't see Cam's work before I did this.

overlookmotel commented on crates/oxc_semantic/src/checker/javascript.rs:
```diff
@@ -106,67 +106,73 @@ fn invalid_let_declaration(x0: &str, span1: Span) -> OxcDiagnostic {
-        if !is_ok {
-            return ctx.error(unexpected_identifier_assign(&ident.name, ident.span));
+            if !is_ok {
+                ctx.error(unexpected_identifier_assign(&ident.name, ident.span));
```
`break` after that statement exits the function. It makes no difference.

overlookmotel committed:
refactor(semantic): refactor `check_binding_identifier` (#12303)

Pure refactor. We had 2 `if` blocks following each other. 1st is `if strict_mode && ...` and the 2nd `if !strict_mode && ...`. Convert into an `if`/`else` instead.
